### PR TITLE
tenantfederation: Simplify code a bit

### PR DESCRIPTION
**Pull request by aknuds1**
#### What this PR does
Simplify `pkg/querier/tenantfederation` code a bit. Based on input from @dimitarvdimitrov.

#### Checklist

- [na] Tests updated
- [na] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/querier/tenantfederation/merge_queryable.go [hunk 1]
```diff
@@ -170,14 +170,12 @@ func (m *mergeQuerier) LabelValues(ctx context.Context, name string, matchers ..
 	spanlog, ctx := spanlogger.NewWithLogger(ctx, m.logger, "mergeQuerier.LabelValues")
 	defer spanlog.Finish()
 
-	matchedTenants, filteredMatchers := filterValuesByMatchers(m.idLabelName, ids, matchers...)
+	matchedIDs, filteredMatchers := filterValuesByMatchers(m.idLabelName, ids, matchers...)
```
Comment (aknuds1) [inline pkg/querier/tenantfederation/merge_queryable.go:173]: Calling it `matchedIDs` is consistent with the naming of the `ids` variable. `matchedIDs` is just a filtered version of `ids`.

**Review by 56quarters — APPROVED** [review]
Only thing I'm concerned about is switching from a slice of IDs (ordered) to a map (not ordered) but it doesn't seem like we were depending on that anywhere :+1:

**Comment by aknuds1** [comment]
> Only thing I'm concerned about is switching from a slice of IDs (ordered) to a map (not ordered) but it doesn't seem like we were depending on that anywhere

Thanks @56quarters. I did actually hit that problem with my first revision (losing ordering),  but then made sure to keep ordering.

**Review by dimitarvdimitrov — COMMENTED** [review]
LGTM, thanks for following up with this @aknuds1

**Pull request merged**